testament/lib/stdtest/testutils.nim

```diff
@@ -27,10 +27,43 @@ template flakyAssert*(cond: untyped, msg = "", notifySuccess = true) =
 
 proc greedyOrderedSubsetLines*(lhs, rhs: string): bool =
   ## returns true if each stripped line in `lhs` appears in rhs, using a greedy matching.
-  let rhs = rhs.strip
-  var currentPos = 0
-  for line in lhs.strip.splitLines:
-    currentPos = rhs.find(line.strip, currentPos)
-    if currentPos < 0:
+  iterator splitLinesLhs(): string {.closure.} =
```

Comment:
* my gut feeling is this is more complicated than it should, but I need to look into it more closely
* how about adding a tests/testament/ttestutils.nim to test this